api/src/server.ts

```diff
@@ -0,0 +1,168 @@
+import express from 'express';
+import bodyParser from 'body-parser';
+
+import morgan from './logging/morgan';
+import winston from './logging/winston';
+
+import session from 'cookie-session';
+import fs from 'fs';
+import https from 'https';
+import cors from 'cors';
+
+import {
+  SECURE_PORT,
+  INSECURE_PORT,
+  KEY_PATH,
+  CERT_PATH,
+  SESSION_SECRET,
+  CORS_OPTIONS,
+  PHP_URI,
+  STATIC_FILE_DIR,
+} from './config';
+
+import { createProxyMiddleware } from 'http-proxy-middleware';
+
+// import routes
+import catalog from './catalog/catalog.routes';
+import cas_auth, {
+  authSoft,
+  authHard,
+  passportConfig,
+} from './auth/cas_auth.routes';
+
+import passport from 'passport';
+
+import * as Sentry from '@sentry/node';
+import * as Tracing from '@sentry/tracing';
+
+Sentry.init({
+  dsn:
+    'https://9360fd2ff7f24865b74e92602d0a1a30@o476134.ingest.sentry.io/5665141',
+
+  // We recommend adjusting this value in production, or using tracesSampler
+  // for finer control
+  tracesSampleRate: 1.0,
+});
+
+// Initialize the app
+const app = express();
+
+// Enable Cross-Origin Resource Sharing
+// (i.e. let the frontend call the API when it's on a different domain)
+app.use(cors(CORS_OPTIONS));
+// Enable request logging.
+app.use(morgan);
+// Enable url-encoding
+app.use(bodyParser.urlencoded({ extended: true }));
+// Trust the proxy.
+// See https://expressjs.com/en/guide/behind-proxies.html.
+app.set('trust proxy', true);
+
+// Strip all headers matching X-COURSETABLE-* from incoming requests.
+app.use((req, _, next) => {
+  for (const header of Object.keys(req.headers)) {
+    // Headers are automatically made lowercase by express.
+    if (header.startsWith('x-coursetable-')) {
+      delete req.headers[header];
+    }
+  }
+  next();
+});
+
+// Redirection routes for historical pages.
```

Comment:
I don't know if there are any more of these, but if you want to make it easier to add new redirects you could just make an obj mapping endpoint to redirect URL and iterate through

Reply:
Yeah so these are the only ones for sure – we retain these for SEO purposes only.